src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.Accessor.cs

```diff
@@ -317,12 +326,16 @@ protected bool GetActualRowIdFromDatabase(SqlConnection connection, Database dat
                 // of those special names, then the use of that name will refer to the declared column
                 // not to the internal ROWID.
                 using var resettableStatement = connection.GetResettableStatement(database == Database.WriteCache
-                    ? _select_rowid_from_writecache_table_where_0
-                    : _select_rowid_from_main_table_where_0);
+                    ? _select_rowid_from_writecache_table_where_0primarykey
+                    : _select_rowid_from_main_table_where_0primarykey);
 
                 var statement = resettableStatement.Statement;
 
-                BindFirstParameter(statement, dataId);
+                // This binds all but the dataNameId primary key parameter.
+                BindPrimaryKeyParameters(statement, dataId);
+                // The data name id parameter is the last in _primaryKeyColumns. So we pass _primaryKeyColumns.Length as
+                // the parameter index as it is 1s based.
+                statement.BindInt64Parameter(parameterIndex: _primaryKeyColumns.Length, dataNameId);
```

Comment:
instead of each accessor having to bind the row-name into the query, it's handled at this single location.